src/contracts/fungible/cache/cache.rs

```diff
@@ -27,6 +29,18 @@ pub trait Cache {
     fn has_asset(&self, id: ContractId) -> Result<bool, Self::Error>;
     fn add_asset(&mut self, asset: Asset) -> Result<bool, Self::Error>;
     fn remove_asset(&mut self, id: ContractId) -> Result<bool, Self::Error>;
+
+    /// Return the map of Utxo-Allocation_amount for a given asset
+    fn utxo_allocation_map(
+        &self,
+        contract_id: ContractId,
+    ) -> Result<BTreeMap<bitcoin::OutPoint, AtomicValue>, Self::Error>;
```

Comment:
Here we will replace the same outpoint if there were multiple assignments to it (see also comments below).

We should use `HashSet<bitcoin::OutPoint, Vec<AtomicValue>>` instead (`HashSet` since we do not need deterministic ordering and the size of the key space will be small)